data_steward/vocabulary.py

```diff
@@ -1,11 +1,179 @@
+
+
+def transform_files(in_dir, out_dir):
+    """
+    Transform vocabulary files in a directory and save result in another directory
+
+    :param in_dir: Directory containing vocabulary csv files
+    :param out_dir: Directory to save the transformed file
+    """
+    fs = os.listdir(in_dir)
+    for f in fs:
+        in_path = os.path.join(in_dir, f)
+        transform_file(in_path, out_dir)
+
+
+def get_aou_general_version():
+    return hash_dir(AOU_GENERAL_PATH)
+
+
+def get_aou_general_vocabulary_row():
+    aou_gen_version = get_aou_general_version()
+    # vocabulary_id vocabulary_name vocabulary_reference vocabulary_version vocabulary_concept_id
+    return DELIMITER.join([AOU_GEN, AOU_GEN, AOU_GEN_VOCABULARY_REFERENCE, aou_gen_version, AOU_GEN_VOCABULARY_CONCEPT_ID])
+
+
+def append_concepts(in_path, out_path):
+    with open(out_path, 'wb') as out_fp:
+        # copy original rows line by line for memory efficiency
+        with open(in_path, 'rb') as in_fp:
+            for row in in_fp:
+                if AOU_GEN in row:
+                    # skip it so it is appended below
+                    warnings.warn(ERROR_APPENDING.format(in_path=in_path))
+                else:
+                    out_fp.write(row)
+
+        # append new rows
+        with open(AOU_GENERAL_CONCEPT_CSV_PATH, 'rb') as aou_gen_fp:
+            _ = next(aou_gen_fp)  # skip header
+            for row in aou_gen_fp:
+                out_fp.write(row)
+
+
+def append_vocabulary(in_path, out_path):
+    new_row = get_aou_general_vocabulary_row()
+    with open(out_path, 'wb') as out_fp:
+        # copy original rows line by line for memory efficiency
+        with open(in_path, 'rb') as in_fp:
+            for row in in_fp:
+                if AOU_GEN in row:
+                    # skip it so it is appended below
+                    warnings.warn(ERROR_APPENDING.format(in_path=in_path))
+                else:
+                    out_fp.write(row)
+        # append new row
+        out_fp.write(new_row)
+
+
+def add_aou_general(in_dir, out_dir):
+    fs = os.listdir(in_dir)
+    concept_in_path = None
+    vocabulary_in_path = None
+    # Case-insensitive search for concept and vocabulary files
+    for f in fs:
+        t, _ = os.path.splitext(f.lower())
+        in_path = os.path.join(in_dir, f)
+        if t == CONCEPT:
+            concept_in_path = in_path
+        elif t == VOCABULARY:
+            vocabulary_in_path = in_path
+    if concept_in_path is None:
+        raise IOError('CONCEPT.csv was not found in %s' % in_dir)
+    if vocabulary_in_path is None:
+        raise IOError('VOCABULARY.csv was not found in %s' % in_dir)
+
+    concept_out_path = os.path.join(out_dir, os.path.basename(concept_in_path))
+    append_concepts(concept_in_path, concept_out_path)
+
+    vocabulary_out_path = os.path.join(out_dir, os.path.basename(vocabulary_in_path))
+    append_vocabulary(vocabulary_in_path, vocabulary_out_path)
+
+
+if __name__ == '__main__':
```

Comment:
I've noticed we have a lot of command line setups like this throughout the pipeline.  Have we ever thought about consolidating/refactoring our environment variables into configuration files, command line options, and the few necessary environment variables?

Reply:
Yes, it has become rather unwieldy. I'd definitely appreciate if you could guide us in our thinking about config files / env vars / options so we can refactor!

Reply:
I think it is something we should plan to improve.  I don't expect us to get it done here and now.  Just wanted to bring it up.